Yank/yank.py

```diff
@@ -245,6 +251,237 @@ def get_region(self, region_name):
+                except ValueError as e:
+                    # Make this a local variable
+                    mdtraj_error = e
+                    mdtraj_output = None
+                else:
+                    mdtraj_error = None
+                return mdtraj_output, mdtraj_error
+
+        class SelectSmarts(Selector):
+
+            @classmethod
+            def select(cls, smarts_string):
+
+                try:
+                    # Skeletal structure, not used yet
+                    raise NotImplementedError("This method has not been implemented yet")
+                except (NotImplementedError, ValueError) as e:
+                    smarts_error = e
+                    smarts_output = None
+                return smarts_output, smarts_error
+
+        class SelectIterable(Selector):
+            # Do not allow these to be sorted as they have been manually specified by integer
+
+            @classmethod
+            def select(cls, iterable):
+                try:
+                    iterable_output = iterable.tolist()
+                except AttributeError:
+                    iterable_output = list(iterable)
+                    iterable_error = None
+                except Exception as e:
+                    iterable_error = e
+                    iterable_output = None
+                else:
+                    iterable_error = None
+                return iterable_output, iterable_error
+
+        class SelectInt(SelectIterable):
+            # Verbatim copy of Iterable, but change name to help error processing
+            pass
+
+        # Dispatcher to parse the selection type and return the valid selection classes
+        @functools.singledispatch
+        def selector_picker(selection_input):
+            try:
+                # Catch both the numpy and built in integers
+                assert all([isinstance(i, np.integer) or isinstance(i, int) for i in selection_input])
+            except AssertionError:
+                raise AssertionError("Selection {} is not iterable of ints or any other readable type such as string! "
+                                     "Unable to parse!".format(selection))
+            return SelectIterable,  # Ensure tuple return
+
+        @selector_picker.register(int)
+        def int_selector(_):
+            return SelectInt,  # Ensure tuple return
+
+        @selector_picker.register(str)
+        def string_selection(_):
+            return SelectRegion, SelectDsl, SelectSmarts
+
+        registered_selectors = selector_picker(selection)
+
+        selector_outputs = []
+        selector_errors = []
+        selector_names = []  # For handling error messages
+        for selector in registered_selectors:
+            selector_names.append(selector.__name__)
+            selection_output, errors = selector.select(selection)
+            selector_outputs.append(selection_output)
+            selector_errors.append(errors)
```

Comment:
The list is populated but unused afterwards.

Reply:
Ah, good catch, this was suppose to be part of a helpful error message, I added that.